src/snovault/elasticsearch/indexer_state.py

```diff
@@ -38,6 +46,100 @@ def includeme(config):
+                    break
+    return instance_name, instance_state
+
+
+def setup_indexing_nodes(request, indexer_state, reset=False):
+    time_now = float(time.time())
+    did_fail = False
+    is_indexing_node = False
+    node_state_template = {
+        'node_index': None,
+        'instance_name': None,
+        'instance_state': None,
+        'waiting_on_remote': False,
+        'started_indexing': False,
+        'done_indexing': False,
+        'last_run_time': str(time_now),
+    }
+    # Get/Setup head node state in local elasticsearch meta data
+    head_node = indexer_state.get_obj(HEAD_NODE_INDEX)
+    indexing_node = indexer_state.get_obj(INDEXING_NODE_INDEX)
+    indexer_tag = '-indexer'
+    # Check which node this is
+    instance_name, instance_state = _get_this_instance_name()
+    if not instance_name:
+        did_fail = True
+        return head_node, indexing_node, time_now, did_fail, is_indexing_node
+    elif not instance_name.replace(indexer_tag, '') == instance_name:
+        # indexing node does not manage states
+        is_indexing_node = True
+        return head_node, indexing_node, time_now, did_fail, is_indexing_node
+    # Handle Setup or Refresh
+    if not head_node or not indexing_node or reset:
+        # Setup node state in elasticsearch
+        # Save this node as head node
+        head_node = copy.deepcopy(node_state_template)
+        head_node['node_index'] = HEAD_NODE_INDEX
+        head_node['instance_name'] = instance_name
+        head_node['instance_state'] = instance_state
+        indexer_state.put_obj(head_node['node_index'], head_node)
+        # Create indexing node
+        indexer_name = f"{instance_name}{indexer_tag}"
+        indexer_instance_name, indexer_instance_state = _get_this_instance_name(
+            instance_name=indexer_name
+        )
+        if not instance_name:
+            did_fail = True
+            return head_node, indexing_node, time_now, did_fail, is_indexing_node
+        indexing_node = copy.deepcopy(node_state_template)
+        indexing_node['node_index'] = INDEXING_NODE_INDEX
+        indexing_node['instance_name'] = indexer_instance_name
+        indexing_node['instance_state'] = indexer_instance_state
+        indexing_node['last_run_time'] = str(float(time.time()))
+        indexer_state.put_obj(indexing_node['node_index'], indexing_node)
+    else:
+        # Refresh aws instance state
+        # Head
+        instance_name, instance_state = _get_this_instance_name(
+            instance_name=head_node['instance_name']
+        )
+        if instance_name and instance_state:
+            head_node['instance_state'] = instance_state
+            indexer_state.put_obj(head_node['node_index'], head_node)
+        # Indexer
+        instance_name, instance_state = _get_this_instance_name(
+            instance_name=indexing_node['instance_name']
+        )
+        if instance_name and instance_state:
```

Comment:
`_get_this_instance_name` might fail, in that case, should `did_fail = True` as we will not update the indexer_state